pkg/apis/workspaces/v1alpha1/commands.go

```diff
@@ -15,12 +15,38 @@ const (
 	CustomCommandType       CommandType = "Custom"
 )
 
+// CommandGroupType describes the kind of command group.
+// +kubebuilder:validation:Enum=build;run;test;debug
+type CommandGroupType string
+
+const (
+	BuildCommandGroupType CommandGroupType = "build"
+	RunCommandGroupType   CommandGroupType = "run"
+	TestCommandGroupType  CommandGroupType = "test"
+	DebugCommandGroupType CommandGroupType = "debug"
+)
+
+type CommandGroup struct {
+	// Kind of group the command is part of
+	Kind CommandGroupType `json:"kind"`
+
+	// +optional
+	// Identifies the default command for a given group kind
+	IsDefault bool `json:"isDefault,omitempty"`
```

Comment:
Just for clarification c.f. #27: we're not enforcing that only one command can be default at the spec level, correct?

Reply:
yes, it was agreed that it would make the overall structure too complicated to express in the specification schema. This will be a constraint mentioned in the specification *in addition to the specification schema*

Reply:
cc @l0rd